### DISCUSSION TITLE
[Question] How to custom template dir?

### DISCUSSION THREAD

**Discussion by kappa-maintainer** [KyoriPowered/blossom#58] [Q&A] [answered]
The default location is good for most of the projects, but some projects decided to use another dir for main source set, in such case changing the template dir would be good.
Btw, could this plugin add Java 8 support? I am using it under a Java 21 template, but some older Minecraft version using FG5+ could use this too

**Comment by zml2008** [comment, marked as answer]
https://github.com/KyoriPowered/blossom/blob/a8002b57fe06d2e026994b731a877f0d93e394b6/src/main/java/net/kyori/blossom/TemplateSet.java#L160 set the template dir here

We have no plans to reduce the minimum java version required for the plugin -- remember that the version of Java that Gradle uses is entirely separate from the version of Java that your project targets, so nothing about Blossom stops you from building projects targeting J8.

**Comment by kappa-maintainer** [comment]
I can change the Java running gradle, but that also make FG using newer JDK to decompile Minecraft source.
And FernFlower somehow will produce different result under different versions of JDK, so setting FG up while using blossom will produce crash.
I don't know if gradle allow override java version for non-compile task or FG allow manually set java version, I just decided to ask here first.
